pkg/apiserver/api_installer.go

```diff
@@ -65,8 +66,16 @@ func (a *APIInstaller) Install() (ws *restful.WebService, errors []error) {
 	redirectHandler := (&RedirectHandler{a.group.Storage, a.group.Codec, a.group.Context, a.info})
 	proxyHandler := (&ProxyHandler{a.prefix + "/proxy/", a.group.Storage, a.group.Codec, a.group.Context, a.info})
 
-	for path, storage := range a.group.Storage {
-		if err := a.registerResourceHandlers(path, storage, ws, watchHandler, redirectHandler, proxyHandler); err != nil {
+	// Register the paths in a deterministic (sorted) order to get a deterministic swagger spec.
+	paths := make([]string, len(a.group.Storage))
+	var i int = 0
+	for path := range a.group.Storage {
```

Comment:
IIUC, That will not work because a.group.Storage is a map. There is no concept of an index in a map.
```
for i, path := range a.group.Storage {
```
will give me the key in i and value in path.

Regarding using
```
paths := []string{}
paths = append(paths, path)
```
Yes. What I have, I believe, is better for allocation.